### ISSUE TITLE
require groups scope bug

### ISSUE THREAD

**Issue by chaffeqa** [bundler/bundler#1448]
I noticed that this Rails snippet in `application.rb` was triggering `require` of everything under `group :development` (triggering requires that caused errors in some gems):

``` .ruby
if defined?(Bundler)
  # If you precompile assets before deploying to production, use this line
  Bundler.require *Rails.groups(:assets => %w(development test))
end
```

This may be a Rails issue, but I noticed it only triggers with **prerelease**, not **stable** Bundler.

Should I notify Rails, or is this a Bundler issue/feature?
# Edit
- [Environment](https://gist.github.com/1246845)
- [Commands that produce error](https://gist.github.com/1246852)

Basically what I did was:
1. Made a Rails 3.1 app
2. Added a gem under `group :development` that produced a backtrace when `required`
3. Ran `bundle exec rails console` to trigger the loading of the **development** env
### The Error

`config/application.rb` runs a line of code (see above) that, as I understand, should `require` the `groups: Rails.env, :default, and :assets` (:assets only if Rails.env= dev/test)

This seems to fly with bundler 1.0, but bundler 1.1 seems to trigger a require on the `group :development` **outside** of the `group :assets`.
